templates/rapidsai-core/partials/install_notebooks.dockerfile.j2

```diff
@@ -7,7 +7,7 @@ RUN gpuci_conda_retry install -y -n rapids \
         "rapids-notebook-env=${RAPIDS_VER}*"
 
 {# Install jupyter lab stuff #}
-RUN gpuci_conda_retry install -y -n rapids jupyterlab-nvdashboard
+RUN gpuci_conda_retry install -y -n rapids jupyterlab-nvdashboard nb_conda_kernels nbgitpuller
```

Comment:
these seem like cool additions, but i have some questions/comments

1) The addition of `nb_conda_kernels` should be fine. However, you've uncovered a bit of tech debt here on our end. We try to keep all the notebook dependencies in the [rapids-notebook-env meta-package in the integration repo](https://github.com/rapidsai/integration/blob/branch-0.18/conda/recipes/rapids-notebook-env/meta.yaml). So ideally `jupyterlab-nvdashboard` and `nb_conda_kernels` would both be added there.

2) Our [notebooks repo](https://github.com/rapidsai/notebooks) consist mostly of a bunch of git-submodules. When I tested the new `gitpuller` command locally, it didn't seem to also clone in the sub-modules (see screenshot below). After exploring the usage section in `gitpuller --help`, I didn't see anything that mentioned submodules so I'm not sure if it's compatible. Also, the `notebooks` repo (like other RAPIDS repos) has a default branch that changes every release, so I'm not sure how this would affect the functionality of `gitpuller`. Any thoughts?

![image](https://user-images.githubusercontent.com/7400326/106800600-a9e7c680-662e-11eb-98d9-7f26d70f9a4c.png)

Reply:
1. Ah ok that's good to know. Should I open a PR over there?
2. Interesting that it doesn't grab the submodules. Looks like there is an open PR to resolve jupyterhub/nbgitpuller#123. Also my intention on using the `${BUILD_BRANCH}` env var was that the correct branch would be pulled.

Reply:
> 1. Ah ok that's good to know. Should I open a PR over there?
> 2. Interesting that it doesn't grab the submodules. Looks like there is an open PR to resolve [jupyterhub/nbgitpuller#123](https://github.com/jupyterhub/nbgitpuller/pull/123). Also my intention on using the `${BUILD_BRANCH}` env var was that the correct branch would be pulled.

1. Yes, if you don't mind that would be great. You could then also remove the entire `RUN gpuci_conda_retry install` line from this PR.
2. Whoops. For this comment, I was referring to this statement in the PR body: `So when we launch the docker image with JupyterHub we can easily update the examples at runtime`. I was thinking about the following for both nightly and stable images:
- **Nightlies** - nightlies are rebuilt every night, so therefore the notebooks are usually up to date when you pull the latest nightly
- **Stable images** - whenever we publish a stable release with new images, the active development branch changes. So if `BUILD_BRANCH` is equal to `branch-0.19` for instance, then after the `0.19` release, all new development will occur on the `branch-0.20` branch. I'm not sure how/if `nbgitpuller` will handle this branch change. I assume (though I didn't confirm this in any way) that `nbgitpuller` will only update the notebooks according to the branch in which it was originally invoked with (i.e. `branch-0.19` in the previous example). If this is the case, then I think that `nbgitpuller` would only be useful in the instance where users download an early nightly container and want to update their notebooks to the latest notebooks _for the RAPIDS version of their current container_ (and not necessarily the latest _overall_ notebooks, which could potentially be on a new branch). Hopefully that makes sense. `nbgitpuller` is small enough that it's probably fine to include in our images regardless, but all of this is moot until they get that submodules PR merged :slightly_smiling_face: .